File: src/main/java/me/bristermitten/plumber/struct/builder/impl/ActionBuilderImpl.java
package me.bristermitten.plumber.struct.builder.impl;

import me.bristermitten.plumber.struct.builder.ActionBuilder;
import me.bristermitten.plumber.struct.builder.TaskLengthConfiguration;
import org.jetbrains.annotations.NotNull;

/**
 * Simple implementation of {@link ActionBuilder} that holds a parent {@link TaskLengthConfiguration}
 *
 * @param <T> the action builder's own type
 */
public abstract class ActionBuilderImpl<T extends ActionBuilder<T>> implements ActionBuilder<T> {
    protected TaskLengthConfiguration<T> parent;

    @NotNull
    @Override
    public TaskLengthConfiguration<T> or() {
        return parent;
    }
}

File: src/main/kotlin/me/bristermitten/plumber/dsl/implementation/ActionBuilderImpl.kt
package me.bristermitten.plumber.dsl.implementation

import me.bristermitten.plumber.dsl.ActionBuilder
import me.bristermitten.plumber.dsl.TaskLengthConfiguration

/**
 * Simple implementation of [ActionBuilder] that holds a parent [TaskLengthConfiguration]
 *
 * @param <T> the action builder's own type
 * */
abstract class ActionBuilderImpl<T : ActionBuilder<T>> : ActionBuilder<T> {
    protected lateinit var parent: TaskLengthConfiguration<T>

    override fun or(): TaskLengthConfiguration<T> {
        return parent
    }
}
